### PR TITLE
Devices names are treated as ephemeral and not persisted to etcd

### PR DESCRIPTION
**Pull request by travisn**
There are a number of changes to clean up devices used by rook:
- Stop using the device name as an etcd key since the names are not reliable. https://github.com/rook/rook/issues/305
- Devices are handled within each node. After we get https://github.com/rook/rook/issues/102, we can have the central orchestration handle requests for adding or removing devices. Until then each node owns dealing with its own osds.
- Only basic device info will be stored in etcd. The detailed info will only be in memory on each machine for osd configuration. The device info is serialized to a single etcd key encoded in json. This simplifies how etcd is updated if devices are added or removed. All you have to do is set a new value
- The first two commits will be squashed before merging upstream. The first commit was already reviewed with https://github.com/rook/rook/pull/287 and only slightly modified to remove the `all` devices option.

### CODE REVIEW

File: pkg/clusterd/inventory/memory.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"fmt"
+	"path"
+	"strconv"
+
+	etcd "github.com/coreos/etcd/client"
+	ctx "golang.org/x/net/context"
+)
+
+func getSystemMemory() uint64 {
+	return uint64(C.sysconf(C._SC_PHYS_PAGES) * C.sysconf(C._SC_PAGE_SIZE))
```
Comment (jbw976) [inline pkg/clusterd/inventory/memory.go]: this is the standard way to get memory size from go code?  Does this new cgo dependency affect our build in any way we need to be aware of?

File: pkg/api/node.go [hunk 2]
```diff
@@ -46,22 +45,11 @@ func (h *Handler) GetNodes(w http.ResponseWriter, r *http.Request) {
...
 		for _, d := range n.Disks {
-			for _, uuid := range appliedIDs {
-				if d.UUID == uuid {
-					// current disk is in applied OSD set, add its storage to the running total
-					storage += d.Size
-				}
-			}
+			// Add up the space of all devices.
+			// We should have a separate metric for osd devices, but keep it simple for now.
+			storage += d.Size
```
Comment (jbw976) [inline pkg/api/node.go]: why do we not care about if we are actually using the disks for storage?  would this value include the storage from a large disk that is formatted and we are not using for the cluster? that would be misleading, but we may have already filtered out those disks that rook won't use somewhere else...not sure.

File: pkg/cephmgr/osd/agent.go [hunk 3]
```diff
@@ -357,6 +350,9 @@ func (a *osdAgent) configureDevices(context *clusterd.Context, devices map[strin
 	go func() {
 		defer connection.Shutdown()
 
+		// set the signal that the osd config is completed
+		defer close(a.osdsCompleted)
```
Comment (jbw976) [inline pkg/cephmgr/osd/agent.go]: what does this osdsCompleted mean?  we close it before we start the loop that does createOrStartOSD?

File: pkg/clusterd/inventory/node_test.go [hunk 4]
```diff
@@ -61,162 +57,64 @@ func TestLoadDiscoveredNodes(t *testing.T) {
...
-
-	return NetworkConfig{
-		Name:        name,
-		IPv4Address: ipv4Addr,
-		IPv6Address: ipv6Addr,
-		Speed:       speed,
-	}
-}
+	// verify the processors
+	assert.Equal(t, 3, len(cfg.Processors))
```
Comment (jbw976) [inline pkg/clusterd/inventory/node_test.go]: do we need to be verifying the properties of the procs in addition to the count?

File: pkg/clusterd/inventory/memory.go (hunk 1 above)
Comment (travisn) [inline pkg/clusterd/inventory/memory.go]: yes, this is the most standard way i could find from Go code to get system memory. I don't see any side effect of adding a dependency

File: pkg/cephmgr/osd/agent.go (hunk 3 above)
Comment (travisn) [inline pkg/cephmgr/osd/agent.go]: this is in a defer statement so it will be executed after createOrStartOSD

File: pkg/api/node.go (hunk 2 above)
Comment (travisn) [inline pkg/api/node.go]: i was thinking in a separate change I would add a "used" flag so we can bubble up this info to the user. But it really is a regression that this isn't working as part of this change, so i'm going to look at this again in this PR.

Comment (travisn) [inline pkg/api/node.go]: after looking into this further, the device usage will be changing with the next issue i'm working on https://github.com/rook/rook/issues/304, so i'll hold off on this for now

**Pull request merged**